pyhap/characteristic.py

```diff
@@ -98,34 +94,66 @@ def __init__(self, display_name, type_id, properties, value=None, broker=None):
             - else use `HAP_FORMAT.DEFAULT` for the format of this characteristic.
         :type value: Depends on `properties["Format"]`
         """
-        assert "Format" in properties and "Permissions" in properties
+        if 'Format' not in properties or 'Permissions' not in properties:
+            raise ValueError('Invalid properties')
         self.display_name = display_name
         self.type_id = type_id
         self.properties = properties
-        if value:
-            self.value = value
-        else:
-            if self.properties.get('ValidValues'):
-                self.value = min(self.properties['ValidValues'].values())
-            else:
-                self.value = HAP_FORMAT.DEFAULT[properties["Format"]]
         self.broker = broker
         self.setter_callback = None
+        self.value = value or self._get_default_value()
 
     def __repr__(self):
         """Return the representation of the characteristic."""
-        return "<characteristic display_name='{}' value={} properties={}>" \
+        return '<characteristic display_name={} value={} properties={}>' \
             .format(self.display_name, self.value, self.properties)
 
+    def _get_default_value(self):
+        """Helper method. Return default value for format."""
+        if self.properties.get('ValidValues'):
+            return min(self.properties['ValidValues'].values())
+        else:
+            value = HAP_FORMAT.DEFAULT[self.properties['Format']]
+            return self._validate_value(value)
+
+    def _validate_value(self, value):
+        """Perform value validation depending on format."""
+        if self.properties.get('ValidValues'):
+            if value not in self.properties['ValidValues'].values():
+                raise ValueError
+            else:
+                return value
+        elif self.properties['Format'] == HAP_FORMAT.STRING:
+            return str(value)[:256]
+        elif self.properties['Format'] == HAP_FORMAT.NUMERIC:
+            value = min(self.properties.get('maxValue', value), value)
```

Comment:
Not every char that is numeric has a max or min value. How about:
min and max -> min
min -> min
max -> max
no min and no max -> (probably handled by ValidValues, but for sake of completeness) 0

Reply:
Actually that's what I should have done here, or at least intended to do (if it doesn't work).
`get('maxValue', value)` has `value` as default. E.g. no maxValue:
 - get --> value
 - min(value, value) -> value